pkgs/top-level/all-packages.nix

```diff
@@ -1919,6 +1919,16 @@ in
     pinentry = if stdenv.isDarwin then pinentry_mac else pinentry;
   };
   gnupg = gnupg21;
+  # suitable to verify signatures non-interactively
+  gnupg-minimal = callPackage ../tools/security/gnupg/21.nix {
```

Comment:
I also thought about copying [21.nix](https://github.com/volth/nixpkgs/blob/master/pkgs/tools/security/gnupg/21.nix) to ```21-minimal.nix```, because gnupg has many files needless in the minimal headless mode (such as localized messages in many languages).
Many installs may result in having both ```gnupg``` and ```gnupg-minimal```.